tools/docker-images/clp-package/build.sh

```diff
@@ -3,12 +3,32 @@
 set -eu
 set -o pipefail
 
+remove_prev_image_and_temp_file() {
+    if [[ -n "$prev_image_id" && "$prev_image_id" != "$new_image_id" ]]; then
+        if docker image inspect "$prev_image_id" >/dev/null 2>&1; then
+            echo "Removing previous image $prev_image_id."
+            docker image remove "$prev_image_id"
+        fi
+    fi
+    rm -f "$temp_iid_file"
+}
```

Comment:
this might be cleaner. though i'm not sure what bash syntaxes / conventions do we prefer
```suggestion
remove_temp_file_and_prev_image() {
    rm -f "$temp_iid_file"

    [[ -z "$prev_image_id" || "$prev_image_id" == "$new_image_id" ]] && return

    if docker image inspect "$prev_image_id" >/dev/null 2>&1; then
        echo "Removing previous image $prev_image_id."
        docker image remove "$prev_image_id"
    fi
}
```